lib/service.go

```diff
@@ -127,32 +127,33 @@ func (m *BoostService) SetFeeRecipientV1(_ *http.Request, args *[]interface{}, r
 
 	// If no relay responded true, return the last error message, or a generic error
 	var err error
-	if !*result {
+	if !result {
 		err = lastRelayError
 		if lastRelayError == nil {
 			err = errors.New("no relay responded true")
 		}
 	}
-	return err
+	return &result, err
 }
 
 // GetHeaderV1 TODO
-func (m *BoostService) GetHeaderV1(_ *http.Request, blockHash *string, result *GetHeaderResponse) error {
+func (m *BoostService) GetHeaderV1(ctx context.Context, blockHash *string) (*GetHeaderResponse, error) {
 	method := "builder_getHeaderV1"
 	logMethod := m.log.WithField("method", method)
 
 	if len(*blockHash) != 66 {
-		return fmt.Errorf("invalid block hash: %s", *blockHash)
+		return nil, fmt.Errorf("invalid block hash: %s", *blockHash)
 	}
 
 	// Call the relay
+	result := new(GetHeaderResponse)
```

Comment:
```suggestion
	var result GetHeaderResponse
```
nitpick: unnecessary allocation, could be replaced by a `var` declaration

Reply:
What would be the difference? `var` also allocates the memory 🤔

Reply:
This is a pointer variable, you’re allocating the value struct unnecessarily

Reply:
Ah my bad, the code suggestion missed the pointer, my bad :)